### PR TITLE
upgrade to isort5

### PR DESCRIPTION
**Pull request by skshetry**
* Imports are no longer moved to the top.
* `seed-isort-config` is no longer needed.
* Has a `black`-compatible profile/config.
* Also `isort`s the imports anywhere in the code
  (not just at the top).

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/plots/data.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def _apply_path(data, path=None, **kwargs):
 def _lists(dictionary):
     for _, value in dictionary.items():
         if isinstance(value, dict):
-            yield from (_lists(value))
+            yield from _lists(value)
```
Comment (skshetry) [inline dvc/repo/plots/data.py:104]: flake8 complained here.

File: dvc/repo/plots/data.py [hunk 2]
```diff
@@ -114,10 +114,12 @@ def _find_data(data, fields=None, **kwargs):
...
-    for l in _lists(data):
-        if all(isinstance(dp, dict) for dp in l):
-            if set(first(l).keys()) & fields == fields:
-                return l
+    for lst in _lists(data):
+        if (
+            all(isinstance(dp, dict) for dp in lst)
+            and set(first(lst).keys()) & fields == fields
+        ):
+            return lst
```
Comment (skshetry) [inline dvc/repo/plots/data.py:122]: I also upgrade flake8, so it complained here which I refactored.

**Comment by skshetry** [comment]
restyled-isort does not have isort5 support. will try adding later.

File: setup.cfg [hunk 3]
```diff
@@ -15,10 +15,6 @@ show_source=true
 count=true
 
 [isort]
-include_trailing_comma=true
+profile=black
 known_first_party=dvc,tests
-known_third_party=PyInstaller,RangeHTTPServer,boto3,colorama,configobj,distro,dpath,flaky,flufl,funcy,git,grandalf,mock,moto,nanotime,networkx,packaging,pathspec,pygtrie,pylint,pytest,requests,ruamel,setuptools,shortuuid,shtab,toml,tqdm,voluptuous,yaml,zc
 line_length=79
```
Comment (skshetry) [inline setup.cfg:20]: @efiop @pared @pmrowla, what do you guys say of increasing line length limit to black's default (i.e. 88)?

**Comment by pmrowla** [comment]
failing CI test is specific to experiments and shouldn't block this PR (see #4401)

**Comment by skshetry** [comment]
> failing CI test is specific to experiments and shouldn't block this PR (see #4401)

I'll rerun later. Thanks. Currently it's blocked by lack of support from restyled.

Keeping this as draft, till restyled-isort supports v5, see https://github.com/restyled-io/restylers/pull/86.

File: setup.cfg (hunk 3 above)
Comment (pmrowla) [reply setup.cfg:20]: Personally, I think 79 is fine. I'm not opposed to using 88, but I would prefer we don't go much higher than that in the future.

IMO Python projects that use 100+ tend to get hard to read, but I get that this is more of a personal taste issue than anything else.

Comment (pared) [reply setup.cfg:20]: I have no strong opinion on that, only thing that bugs me is that this decision will require some refactor, as black tends to not remove `"` when its not needed anymore, so we will probably have a lot of lines like
`"This is " "the comment"`

Comment (skshetry) [reply setup.cfg:20]: @pared, string handling has been significantly improved on black (https://github.com/psf/black/pull/1132) and should solve that issue. Though that'd be useful only if there were a new release. :(

**Comment by skshetry** [comment]
PR has been merged and the new `restyled-isort` image is available. :tada:

File: setup.cfg (hunk 3 above)
Comment (skshetry) [reply setup.cfg:20]: Okay then, if no one has *strong* opinion, I'll be increasing the limit to 88.

File: tests/remotes/__init__.py [hunk 4]
```diff
@@ -35,6 +35,14 @@
...
 
 
+TEST_REMOTE = "upstream"
+TEST_CONFIG = {
+    "cache": {},
+    "core": {"remote": TEST_REMOTE},
+    "remote": {TEST_REMOTE: {"url": ""}},
+}
+
+
```
Comment (skshetry) [inline tests/remotes/__init__.py:45]:
```suggestion
```

Rebase duplicated this, removing

File: setup.cfg (hunk 3 above)
Comment (efiop) [reply setup.cfg:20]: Oops, missed this one. I would not increase it to 88. 80 is a good standard width that is used in many projects and is on the border of being revieable in github split screen view for PRs. Plus, if we change it now, we'll introduce inconsistency with the rest of the code base. So I'm not really seeing good reasons to go to 88, it won't make it easier to navigate.

**Pull request merged**